### PR TITLE
Fix deprecated GLPI methods calls

### LINKED ISSUE #365: getFromDBByQuery deprecated in GLPI 9.3
at least one PR fails with GLPI 9.3 (currently branch master) because getFromDBByQuery is deprecated.

We need to find all calls to this method and replace them by $DB->request(), in order to make tests work again. 

This is high priority because other tests on other PRs depend on this (and not so hard)

### PR DESCRIPTION
**Pull request by btry**
fix #365 

This PR requires https://github.com/glpi-project/glpi/pull/3724

### CODE REVIEW

File: inc/agent.class.php [hunk 1]
```diff
@@ -928,11 +930,23 @@ public function getByTopic($topic) {
             if (strlen($serial)) {
                $computerTable = Computer::getTable();
                $agentTable = self::getTable();
-               return $this->getFromDBByQuery("LEFT JOIN `$computerTable` `c` ON (
-                     `c`.`id` = `$agentTable`.`computers_id`
-                     )
-                     WHERE `$agentTable`.`entities_id`='$entity' AND `c`.`serial` = '$serial'"
-               );
+               $result = $DB->request([
```
Comment (DIOHz0r) [inline inc/agent.class.php]: You forgot to add the WHERE arguments on the new request

File: inc/config.class.php [hunk 2]
```diff
@@ -188,7 +188,7 @@ public function addDocumentTypes() {
 
       foreach ($extensions as $extension => $name) {
          $documentType = new DocumentType();
-         if (!$documentType->getFromDBByQuery("WHERE LOWER(`ext`)='$extension'")) {
+         if (!$documentType->getFromDBByCrit(['ext' => $extension])) {
```
Comment (DIOHz0r) [inline inc/config.class.php]: Doesn't the lower matters?

Comment (btry) [reply inc/config.class.php]: well... no. Mysql is case insensitive when comparing strings !

Try to select an entry of glpi_documenttypes with a where on the column ext, and with an existing entry you type un upper case...

File: inc/agent.class.php (hunk 1 above)
Comment (btry) [reply inc/agent.class.php]: +1

File: inc/agent.class.php [hunk 3]
```diff
@@ -918,26 +920,34 @@ public function getTopic() {
...
+               ]
+            ]
+         ],
+         'WHERE' => [
+            'AND' => [
+               PluginFlyvemdmAgent::getTable() . '.' . Entity::getForeignKeyField() => $entity,
+               Computer::getTable() . '.serial' => $serial
+            ]
+         ]
+      ]);
```
Comment (btry) [inline inc/agent.class.php]: refactored to make code easier to read by reversing the if()

**Pull request merged**